### ISSUE TITLE
domain.analyze reporting "invalid" for valid domains

### ISSUE THREAD

**Issue by berzniz** [hapijs/address#29]
#### Support plan

* *which support plan is this issue covered by?* Community
* *is this issue currently blocking your project?* no
* *is this issue affecting a production system?* yes

#### Context

* *node version*: any (but checked on 12.16.3 LTS)
* *module version with issue*: 4.0.1
* *last module version without issue*: none
* *environment* (e.g. node, browser, native): node
* *used with* (e.g. hapi application, another framework, standalone, ...):standalone
* *any other relevant information*: no

#### What are you trying to achieve or the steps to reproduce?

I am trying to validate the following email provided by a customer (`hello@1global`). While not common, it is a valid email without a tld that starts with a digit. This is how the domain is verified:
```js
domain.analyze('1global', {minDomainSegments: 1, tlds: false})
// { error: 'Domain contains invalid character' }
```

#### What was the result you got?
The domain `1global` was invalid, and therefore the email `hello@1global` was invalid as well

#### What result did you expect?
I expected the domain `1global` to be valid with the provided options, and therefore the email `hello@1global` to be valid as well.

**Comment by hueniverse** [comment]
The domain is not valid according to the spec. TLD cannot start with a number. Also, for this to be value, `1global` would have be a valid registered TLD to be resolved.

**Comment by berzniz** [comment]
Hi Eran, I really appreciate the response and your explanation. I'll dive deeper into the spec to better understand it and see why my claim was incorrect. I read a [summary of related RFCs](https://stackoverflow.com/a/53875771/269666) which concluded it could be valid, but it could be wrong interpretation.

I'll have to verify it with a non-email validator (using Joi), as other systems we rely on (in our case the SaaS platform of HubSpot), will consider these to be valid and send it our way.

**Comment by hueniverse** [comment]
Are you able to actually send email to that address?

**Comment by berzniz** [comment]
I believe you are correct as Gmail will not allow to send to `hello@1global`:
![image](https://user-images.githubusercontent.com/170214/82658595-ee157100-9c2f-11ea-9ea7-17d265321956.png)

However, it doesn't allow to send to `hello@global` which `@hapi/address` says is valid (under a specific configuration):
![image](https://user-images.githubusercontent.com/170214/82658655-11402080-9c30-11ea-8763-21b41121bedd.png)

```
address.email.isValid('hello@global', {minDomainSegments: 1, tlds: false})
// true
```

So if `hello@global` is valid, should `hello@1global` be valid as well? It seems both could have had the same output (Google declares both are invalid).

Thanks for your time looking at this, I agree this is not worth researching further as this is a very edge use case. We just had to deal with it as an upstream provider "breaks" the email rules by being more relaxed and we had to relax those as well.
